test/unit/foreman/renderer_test.rb

```diff
@@ -327,6 +327,56 @@ def setup_safemode_renderer
     assert @renderer.host_param_false?('false_param')
   end
 
+  test 'should have subnet_has_param?' do
+    @renderer.host = FactoryGirl.create(:host, :with_puppet)
+    subnet = FactoryGirl.create(:subnet_ipv4)
+    subnet.subnet_parameters.create(name: 'myparam', value: 'myvalue')
+    assert @renderer.subnet_has_param?(subnet, 'myparam').present?
+  end
+
+  test 'should have subnet_has_param? helper returning true' do
+    @renderer.host = FactoryGirl.create(:host, :with_puppet)
+    subnet = FactoryGirl.create(:subnet_ipv4)
+    subnet.subnet_parameters.create(name: 'myparam', value: 'myvalue')
+    assert @renderer.subnet_has_param?(subnet, 'myparam') == true
+  end
+
+  test 'should have subnet_has_param? helper returning false' do
+    @renderer.host = FactoryGirl.create(:host, :with_puppet)
+    subnet = FactoryGirl.create(:subnet_ipv4)
+    subnet.subnet_parameters.create(name: 'myparam', value: 'myvalue')
+    assert @renderer.subnet_has_param?(subnet, 'my_wrong_param') == false
+  end
+
+  test 'should have subnet_has_param? helper returning false when subnet is nil' do
+    @renderer.host = FactoryGirl.create(:host, :with_puppet)
+    exception = assert_raises Foreman::Renderer::WrongSubnetError do
+      @renderer.subnet_has_param?(nil, 'myparam') == false
+    end
+    assert_include exception.message, 'expected a subnet'
+  end
+
+  test 'should render existing subnet param using "subnet_param" helper' do
+    @renderer.host = FactoryGirl.create(:host, :with_puppet)
+    subnet = FactoryGirl.create(:subnet_ipv4)
+    subnet.subnet_parameters.create(name: 'myparam', value: 'myvalue')
+    assert_equal @renderer.subnet_param(subnet, 'myparam'), 'myvalue'
+  end
+
+  test 'should not render missing subnet param using "subnet_param" helper' do
+    @renderer.host = FactoryGirl.create(:host, :with_puppet)
+    subnet = FactoryGirl.create(:subnet_ipv4)
+    subnet.subnet_parameters.create(name: 'myparam', value: 'myvalue')
+    assert @renderer.subnet_param(subnet, 'my_wrong_param').present? == false
```

Comment:
same as above, parameter creation seems unnecessary and this line would actually be better as

`assert_nil @renderer.subnet_param(subnet, 'my_wrong_param')`